artpaint/paintwindow/PaintWindow.cpp

```diff
@@ -130,57 +131,31 @@ PaintWindow::PaintWindow(BRect frame, const char* name, uint32 views,
 		fAdditionalHeight += fMenubar->Bounds().Height() + 1.0;
 	}
 
-	BRect bounds = Bounds().InsetByCopy(-1.0, -1.0);
-	bounds.top = fMenubar ? fMenubar->Bounds().Height() : bounds.top;
-
-	fHorizontalScrollbar = new BScrollBar(BRect(bounds.left,
-		bounds.bottom - B_H_SCROLL_BAR_HEIGHT,
-		bounds.right - B_V_SCROLL_BAR_WIDTH, bounds.bottom),
-		"horizontal", NULL, 0, 0, B_HORIZONTAL);
-	AddChild(fHorizontalScrollbar);
+	fHorizontalScrollbar = new BScrollBar(BRect(0, 0, 0, 0), "horizontal", NULL, 0, 0, B_HORIZONTAL);
 	fHorizontalScrollbar->SetRange(0.0, 0.0);
 	fHorizontalScrollbar->SetSteps(8.0, 32.0);
 
-	fVerticalScrollbar = new BScrollBar(BRect(bounds.right - B_V_SCROLL_BAR_WIDTH,
-		bounds.top, bounds.right, bounds.bottom - B_H_SCROLL_BAR_HEIGHT),
-		"vertical", NULL, 0, 0, B_VERTICAL);
-	AddChild(fVerticalScrollbar);
+	fVerticalScrollbar = new BScrollBar(BRect(0, 0, 0, 0), "vertical", NULL, 0, 0, B_VERTICAL);
 	fVerticalScrollbar->SetRange(0.0, 0.0);
 	fVerticalScrollbar->SetSteps(8.0, 32.0);
 
 	fAdditionalWidth += B_V_SCROLL_BAR_WIDTH - 1.0;
 	fAdditionalHeight += B_H_SCROLL_BAR_HEIGHT - 1.0;
 
-	bounds.top += 1.0;
-	bounds.right -= B_V_SCROLL_BAR_WIDTH + 1.0;
-	bounds.bottom -= B_H_SCROLL_BAR_HEIGHT + 1.0;
-
 	// The status-view is not optional. It contains sometimes also buttons that
 	// can cause some actions to be taken.
 	if ((views & HS_STATUS_VIEW) != 0) {
 		// Create the status-view and make it display nothing
 		// Note: this sucks, since the status view resizes himself...
-		fStatusView = new StatusView(BRect(0.0, bounds.bottom, bounds.right,
-			bounds.bottom));
+		fStatusView = new StatusView(BRect(0, 0, 0, 0));
 		fStatusView->DisplayNothing();
 
-		// place the statusbar along bottom of window on top of scrollbar
-		// and also record size that it occupies
-
-		// here add the statusview to window's hierarchy
-		AddChild(fStatusView);
-
-		// update the bottom value to be 1 pixel above status view
-		bounds.bottom -= fStatusView->Bounds().Height() + 1.0;
-
 		// update the fAdditionalHeight variable
 		fAdditionalHeight += fStatusView->Bounds().Height() + 1.0;
 	}
 
 	// make the background view (the backround for image)
-	fBackground = new BackgroundView(BRect(0.0, bounds.top, bounds.right,
-		bounds.bottom));
-	AddChild(fBackground);
+	fBackground = new BackgroundView(BRect(0, 0, 0, 0));
```

Comment:
OK. Let's keep tis PR open for a day should @waddlesplash wishes to comment. Otherwise, @waddlesplash, feel free to merge.